### PR TITLE
WIP: add preprocess function

### PR DESCRIPTION
**Pull request by scheidec**

### CODE REVIEW

**Comment by scheidec** [comment]
@MBattagl See readme for usage.  I still have to add tests but wanted to start the review process to make sure everything that needs to be handled here is and the output is in the right place to pipe into the `descriptives` function to be able to utilize the labels.

File: README.md [hunk 1]
```diff
@@ -16,9 +16,156 @@ devtools::install_github("MethodsConsultants/tibbletest")
...
+    ## -------------  ------
+    ## gender               
+    ## age                  
+    ## sugar_factor         
+    ## treat                
+    ## happiness            
+    ## happy                
+    ## weight               
+    ## no_weight            
+    ## treat2
```
Comment (MBattagl) [inline README.md]: Doesn't look to currently work if specifying your own labels

Comment (MBattagl) [reply README.md]: "## Argument `label` must be of same length as numbers of columns in `x`."

Comment (scheidec) [reply README.md]: forgot to reknit

**Comment by MBattagl** [comment]
The formatting of the attributes is different when using the default vs passing in `var_labels`

```
library(tibbletest)

# defaults
example_dat %>% 
  dplyr::rename(`sugar   factor` = sugar_factor) %>% 
  preprocess() %>% 
  str()

# non-defaults
mylabels <- list(
  gender           = "Sex",
  sugar_factor     = "Sugar Amount",
  treat2           = "Alternative Treatment"
)

example_dat %>% 
  dplyr::rename(`sugar   factor` = sugar_factor) %>% 
  preprocess(clean_names = TRUE, convert_to_logical = FALSE, var_labels = mylabels) %>%
  str()
```

**Comment by scheidec** [comment]
I'm assuming we'd prefer the attributes that result from running with the defaults? This way both the cleaned name and label are stored?

**Comment by MBattagl** [comment]
What is the default when using `sjlabelled::set_label`? Also what is the default when reading in SPSS or Stata data files? It would be nice to have our label attributes work the same way as when reading in with `haven`, so downstream functions don't need to handle multiple formats

**Comment by scheidec** [comment]
The default behavior of `preprocess` is just using `sjlabelled::set_label`, so those are the attributes that function assigns by default.

when `haven` reads in SPSS and Stata files it assigns a special attribute for each file type:

```
haven::read_sav("mtcars.sav") %>% str()
Classes ‘tbl_df’, ‘tbl’ and 'data.frame':	32 obs. of  11 variables:
 $ mpg : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ cyl : num  6 6 4 6 8 6 8 4 4 6 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ disp: num  160 160 108 258 360 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ hp  : num  110 110 93 110 175 105 245 62 95 123 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ drat: num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ wt  : num  2.62 2.88 2.32 3.21 3.44 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ qsec: num  16.5 17 18.6 19.4 17 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ vs  : num  0 0 1 1 0 1 0 1 1 1 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ am  : num  1 1 1 0 0 0 0 0 0 0 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ gear: num  4 4 4 3 3 3 3 4 4 4 ...
  ..- attr(*, "format.spss")= chr "F8.2"
 $ carb: num  4 4 1 1 2 1 4 2 2 4 ...
  ..- attr(*, "format.spss")= chr "F8.2"
```

```
haven::read_dta("mtcars.dta") %>% str()
Classes ‘tbl_df’, ‘tbl’ and 'data.frame':	32 obs. of  11 variables:
 $ mpg : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ cyl : num  6 6 4 6 8 6 8 4 4 6 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ disp: num  160 160 108 258 360 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ hp  : num  110 110 93 110 175 105 245 62 95 123 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ drat: num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ wt  : num  2.62 2.88 2.32 3.21 3.44 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ qsec: num  16.5 17 18.6 19.4 17 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ vs  : num  0 0 1 1 0 1 0 1 1 1 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ am  : num  1 1 1 0 0 0 0 0 0 0 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ gear: num  4 4 4 3 3 3 3 4 4 4 ...
  ..- attr(*, "format.stata")= chr "%10.0g"
 $ carb: num  4 4 1 1 2 1 4 2 2 4 ...
```

**Comment by scheidec** [comment]
we'll clearly need to test the behavior of our functions on files read in from haven to make sure those work properly

**Comment by MBattagl** [comment]
I'm actually working with a `spss` file right now that has variable labels. Here's how they look:

```
 $ parent_gender                                     : num  1 2 2 1 1 2 2 2 2 2 ...
  ..- attr(*, "label")= chr "Parent/Caregiver’s Gender (check one):"
 $ parent_race                                       : num  7 2 2 2 2 2 2 2 2 2 ...
  ..- attr(*, "label")= chr "Parent/Caregiver’s Race (check all that apply):"
 $ child_gender                                      : num  1 1 1 1 1 2 2 2 2 2 ...
  ..- attr(*, "label")= chr "Child/Youth’s Gender"
 $ child_ethnicity                                   : num  2 2 NA NA NA 2 2 2 2 NA ...
  ..- attr(*, "label")= chr "Child/Youth’s Ethnicity"
 $ child_race                                        : num  6 2 8 6 4 2 NA 2 2 7 ...
  ..- attr(*, "label")= chr "Child/Youth’s Race"
 $ start_child_cws                                   : num  2 2 4 2 4 4 NA 2 4 2 ...
  ..- attr(*, "label")= chr "What is the status of this child/youth’s involvement with the child welfare system"
 $ start_child_jj                                    : num  1 1 1 1 1 1 1 3 1 1 ...
  ..- attr(*, "label")= chr "What is the status of this child/youth’s involvement with the juvenile justice system"
```

**Pull request closed without merging**